File: polyglot-scala-impl/src/main/scala/org/realsoftwarematters/polyglot/scala/Twitter.scala
package org.realsoftwarematters.polyglot.scala

import java.util.List
import java.util.ArrayList

import org.realsoftwarematters.polyglot.ITwitter
import org.realsoftwarematters.polyglot.Twitt

class Twitter extends ITwitter {

	var twitts : List[Twitt] = new ArrayList()
	
	def addTwitt(t:Twitt) = {twitts.add(t)}

	def getAll : List[Twitt] = twitts
	
	def getImplementation : String = "scala"
	
}

File: polyglot-java-impl/src/main/java/org/realsoftwarematters/polyglot/java/Twitter.java
package org.realsoftwarematters.polyglot.java;

import java.util.ArrayList;
import java.util.List;

import org.realsoftwarematters.polyglot.ITwitter;
import org.realsoftwarematters.polyglot.Twitt;

public class Twitter implements ITwitter{

	private List<Twitt> twitts;
	
	public Twitter(){
		twitts = new ArrayList<Twitt>();
	}
	
	public void addTwitt(Twitt t) {
		twitts.add(t);
	}

	public List<Twitt> getAll() {
		return twitts;
	}

	public String getImplementation() {
		return "Java";
	}

	
}
